src/Microsoft.Extensions.Http.Polly/PolicyHttpMessageHandler.cs

```diff
@@ -77,7 +92,13 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
                 throw new ArgumentNullException(nameof(request));
             }
 
-            return _policy.ExecuteAsync((ct) => base.SendAsync(request, ct), cancellationToken);
+            var policy = ResolvePolicy(request);
+            return policy.ExecuteAsync((ct) => base.SendAsync(request, ct), cancellationToken);
```

Comment:
Should there be a null-check on `policy` after `ResolvePolicy()` returns?

Reply:
I agonized over this and couldn't really come to a good decision. Usually my philosophy is *play stupid games, win stupid prizes*... 

This can be done in a way that doesn't impact the perf of the non-dynamic way so it's probably worth doing to allay confusion.